src/error.rs

```diff
@@ -2,8 +2,9 @@ use std::io::Error as IoError;
 use std::result;
 use std::fmt;
 use std::error::Error as StdError;
+use std::net::TcpStream;
 
-use openssl::ssl::error::SslError;
+use openssl::ssl::HandshakeError as SslError;
```

Comment:
That is correct but because the `impl Client` abstracts over `Read+Write` a other ssl related error will never reach us. All we get is an `std::io::Error` . The only ssl related Error we get is at the start of the connection.

I have a diff where I add the more generic `SslError` to the `Error` enum but it is kind of useless because it is never used. If you want i can add it.